src/input/input_python.rs

```diff
@@ -339,6 +344,8 @@ impl<'a> Input<'a> for PyAny {
             decimal_as_int(self.py(), self, decimal)
         } else if let Ok(float) = self.extract::<f64>() {
             float_as_int(self, float)
+        } else if let Some(enum_val) = maybe_as_enum(self) {
+            Ok(EitherInt::Py(enum_val))
```

Comment:
@davidhewitt I'd like to get your thoughts but I fear this would slow down all int parsing error cases right?

Reply:
Seems like it. Any suggestions on a cheaper bypass check?

Reply:
@adriangb I think I fixed the performance issue by using `interned` strings, so we avoid unnecessary allocations. Would you mind taking another look?

Reply:
As far as performance goes I don't think `maybe_as_enum` is very expensive; apart from a one-time import of the `enum.EnumMeta` object the `.get_type()` and `.is()` methods are just pointer operations so should be extremely cheap.